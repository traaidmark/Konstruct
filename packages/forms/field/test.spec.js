// FORM [ FIELD > TESTS ] ######################################################

// 1. DEPENDENCIES +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 1.1. EXTERNAL DEPENDENCIES ..................................................

import React from 'react';
import renderer from 'react-test-renderer';

// 1.1. END ....................................................................

// 1.2. INTERNAL DEPENDENCIES ..................................................

import Field from './src';

// 1.2. END ....................................................................

// 1. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 2. PROPS ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

const props = {
  type: 'text',
  label: 'field label',
  name: 'field',
  placeholder: 'enter your details',
  value: '01',
  onChange: () => console.log('lol'),
  prefix: 'zar',
  postfix: 'zar',
  isDisabled: false,
  message: 'This is a test message',
  hasError: false,
  required: false,
  size: 'default',
}

// 2. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 3. TEST +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

describe('Form: Field', () => {

  // 3.1. RENDER WITHOUT PROPS .................................................

  test('renders correctly without any props', () => {
    const tree = renderer.create(
      <Field />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // 3.1. END ..................................................................

  // 3.2. RENDER WITH ALL PROPS ................................................

  test('renders correctly with all props', () => {
    const tree = renderer.create(
      <Field { ...props } />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // 3.1. END ..................................................................

});

// 3. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// END OF FILE #################################################################
