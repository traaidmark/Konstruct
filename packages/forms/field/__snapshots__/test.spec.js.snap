// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form: Field renders correctly with all props 1`] = `
<div
  className="form-field "
>
  <label
    className="form-field-label"
    htmlFor="field"
  >
    field label
  </label>
  <span
    className="form-field-prefix"
  >
    zar
  </span>
  <input
    className=""
    disabled={false}
    id="field"
    name="field"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="enter your details"
    required={false}
    type="text"
    value="01"
  />
  <span
    className="form-field-postfix"
  >
    zar
  </span>
  <small
    className="form-field-message"
  >
    This is a test message
  </small>
</div>
`;

exports[`Form: Field renders correctly without any props 1`] = `
<div
  className="form-field "
>
  <input
    className=""
    disabled={false}
    onChange={[Function]}
    onKeyDown={[Function]}
    required={false}
    type="text"
  />
</div>
`;
